pkg/libvirttools/virtualization.go

```diff
@@ -277,6 +280,58 @@ func (v *VirtualizationTool) addVolumesToDomain(podID, podName string, domain *l
 	return nil
 }
 
+func vmLogToFile() bool {
+	if os.Getenv("VIRTLET_VM_LOG_TO_FILE") != "" {
+		return true
+	}
+	return false
+}
+
+func (v *VirtualizationTool) RemoveLibvirtSandboxLog(sandboxId string) error {
+	return os.RemoveAll(filepath.Join(vmLogBaseDir, sandboxId))
+}
+
+func (v *VirtualizationTool) addSerialDevicesToDomain(sandboxId string, domain *libvirtxml.Domain, settings VirtletDomainSettings) error {
+	if settings.vmLogToFile {
+		logDir := filepath.Join(vmLogBaseDir, sandboxId)
+		logPath := filepath.Join(logDir, vmLogFileName)
+
+		// Prepare directory where libvirt will store log file to.
+		if err := os.Mkdir(logDir, 0644); err != nil {
+			glog.Warningf("failed to create vmLogDir '%s': %s", logDir, err.Error())
```

Comment:
There is a story behind this one: if I was to return error here then if someone attempted to put two VMs in same Pod, it failed here for the second VM. The error was "directory already exists". So I downgraded it into warning so that Virtlet didn't crash when directory already existed.

But you are right, error should be raised here - preventing user to put two VMs into the same Pod should be prohibited. Resolved by only attempting to create directroy if it does not exist yet.